assets/app/hooks/diff_pulse.js

```diff
@@ -0,0 +1,31 @@
+const DiffPulse = {
+  mounted() {
+    if (this.el.hasAttribute('data-pulse')) {
+      this.el.removeAttribute('data-pulse');
+
+      this.el.classList.add('animate-diff-pulse');
+
+      this.timeout = setTimeout(() => {
+        this.el.classList.remove('animate-diff-pulse');
+      }, 500);
+    }
+  },
+  updated() {
+    if (this.el.hasAttribute('data-pulse')) {
+      clearTimeout(this.timeout);
+      this.el.removeAttribute('data-pulse');
+      this.el.classList.remove('animate-diff-pulse');
+
+      setTimeout(() => this.el.classList.add('animate-diff-pulse'));
+
+      this.timeout = setTimeout(() => {
+        this.el.classList.remove('animate-diff-pulse');
+      }, 500);
+    }
+  },
+  destroyed() {
+    clearTimeout(this.timeout);
+  },
+};
```

Comment:
Can't we do it like this? 
```suggestion
  mounted() {
    if (this.el.classList.contains("animate-diff-pulse")) {
      this.timeout = setTimeout(() => {
        this.el.classList.remove('animate-diff-pulse');
      }, 500);
    }
  },
  updated() {
    if (this.el.classList.contains("animate-diff-pulse")) {
      clearTimeout(this.timeout);
      
      this.timeout = setTimeout(() => {
        this.el.classList.remove('animate-diff-pulse');
      }, 500);
    }
  },
  destroyed() {
    clearTimeout(this.timeout);
  },
};
```

Reply:
When assigns change in intervals shorter than the time of this animation it has to be reset manually to look properly. It might look confusing but this is what I came up with to achieve proper effect.